File: Server/Card.cpp
#include "Card.h"


Card::Card(unsigned short values[15]) {
	
	for (int i=0; i<3; ++i)
	{
		this->completeLine[i] = false;

		for (int j=0; j<5; ++j)
		{
			this->notMarked[i].push_back(values[j + (i*5)]);
			this->values[j+(i*5)] = values[j + (i*5)];
		}
	}
}


Card::~Card() { }


unsigned short Card::EvaluateBall (unsigned short ball)
{
	unsigned short retValue = 0;        // 0 -> nothing special, 1 -> LINE!, 2 -> BINGO!!
	for (int i=0; i<3; ++i)
	{
		if(std::find(this->notMarked[i].begin(), this->notMarked[i].end(), ball) != this->notMarked[i].end())
		{
			this->notMarked[i].remove(ball);
			this->marked[i].push_back(ball);
			if (this->marked[i].size() == 5)
			{
				completeLine[i] = true;
				retValue = 1;
			}

			if (this->completeLine[0] && this->completeLine[1] && this->completeLine[2])
			{
				retValue = 2;
			}
		}
	}

	return retValue;
}


std::string Card::StringCode ()
{
	std::string code ;
	
	for (int i=0; i<15; ++i)
	{
		code += this->values[i] + '0';
	}

	return code;
}



std::string Card::SStringCode (unsigned short values[15]){
	std::string code ;
	
	for (int i=0; i<15; ++i)
	{
		code += values[i] + '0';
	}

	return code;
}

File: Client/Assets/NetworkLib/Card.cs
using System.Collections.Generic;

public class Card
{
    public ushort ID { get; protected set; }
    public ushort[] values;
    public List<ushort>[] notMarked;
    public List<ushort>[] marked;
    public bool[] completeLine;

    public Card (ushort [] values)
    {
        this.values = values;
        notMarked = new List<ushort>[3];
        marked = new List<ushort>[3];
        completeLine = new bool[3];

        for (int i = 0; i < 3; ++i)
        {
            notMarked[i] = new List<ushort>();
            marked[i] = new List<ushort>();
            completeLine[i] = false;

            for (int j = 0; j < 5; ++j)
            {
                notMarked[i].Add(values[j + (i * 5)]);
            }
        }
    }

    public ushort EvaluateBall (ushort ball)
    {
        ushort retValue = 0;        // 0 -> nothing special, 1 -> LINE!, 2 -> BINGO!!
        for (int i = 0; i < 3; ++i)
        {
            if (notMarked[i].Contains(ball))
            {
                notMarked[i].Remove(ball);
                marked[i].Add(ball);
                if (marked[i].Count == 5)
                {
                    completeLine[i] = true;
                    retValue = 1;
                }

                if (completeLine[0] && completeLine[1] && completeLine[2])
                {
                    retValue = 2;
                }
            }
        }

        return retValue;
    }

    public string StringCode ()
    {
        string code = values[0].ToString();
        
        for (int i=1; i<15; ++i)
        {
            code += ("-" + values[i].ToString());
        }

        return code;
    }
}
